moveit_ros/planning/rdf_loader/include/moveit/rdf_loader/synchronized_string_parameter.h

```diff
@@ -80,5 +84,7 @@ class SynchronizedStringParameter
   rclcpp::Publisher<std_msgs::msg::String>::SharedPtr string_publisher_;
 
   std::string content_;
+  boost::lockfree::spsc_queue<std::string> queue_;
+  std::mutex queue_mutex_;
```

Comment:
Is there a need for a mutex? I thought lock-free mean no mutex needed

Reply:
The reason for the lock is that pushing into a spsc queue is not re-entrant safe.  Because there are many threads that can be run by a ROS executor and there is the possibility of the callback being called on two threads at once you need to protect pushing into the spsc_queue.